### ISSUE TITLE
Android tv

### ISSUE THREAD

**Issue by city57** [TeamNewPipe/NewPipe#1363]
Hello(i french)

I would like to use newpipe on Android TV, I installed it and it works fine! Except at the launch of the video. The video changes position!
I have to use VLC and it's a shame ....
Newpipe could be a reference for Android TV to replace youtube TV

**Comment by theScrabi** [comment]
We once thought and worked about this, but the AndroidTV port is currently on ice.

**Comment by city57** [comment]
Too bad it works well with VLC player. It is just the default player that
changes the position of the image.
In any thanks for your application me I put on my mi-box Xiaomi  with VLC

**Comment by theScrabi** [comment]
@wb9688 maybe instead of making NewPipe work with leanback we could just focuse on Making it work with arrow keys, so it can be used with a remote control. This would generate a usable outcome with much less of effort, and it would be a valuable beginning. What would you think?

**Comment by city57** [comment]
It is likely that not everyone has a "all in one Microsoft" keyboard. Yes a
use by arrow remote control is necessary. However it is imperative to
adjust the screen position change bug first.

**Comment by theScrabi** [comment]
What do you meen by adjust the screen position?

**Comment by gabriellluz** [comment]
VLC has a Android TV feature. Maybe we could use that as a reference.

**Comment by theScrabi** [comment]
Yes but vlc for android is based on the lean back UI, and we can not support lean back by now.

**Comment by city57** [comment]
I solved the  problem with landscape Android

https://drive.google.com/file/d/1wywWY4dBVIW7WJUg8ddcoR0lC4ovUyjt/view

But it's a pity

**Comment by theScrabi** [comment]
@city57 I can't test it right now. Is that a modified version of NewPipe?

**Comment by aiddroid** [comment]
Works fine on MiTV(xiaomi TV, android 6.0), but cannot show player controller while playing the video.So I can't fastward or  change speed via the TV remoter, please fix it. thanks.

**Comment by PoliNetwork** [comment]
I think this issue is still open

**Comment by Alexander-TX** [comment]
I have uploaded a branch with some fixes for TV devices: https://github.com/Alexander-TX/NewPipe/tree/eltex-tv-fixes

It is far from proper adaptation, but at least all basic functionality (video playback, comments) is accessible via directional navigation.

In addition there is a special overlay that better highlights currently focused element.

**Comment by Stypox** [comment]
@Alexander-TX you could open a (possibly draft) pull request, so discussion can be kept there. Make sure to @ mention everyone here so that they can express their opinions and test.

**Comment by sorryusernameisalreadytaken** [comment]
> 
> 
> I have uploaded a branch with some fixes for TV devices: https://github.com/Alexander-TX/NewPipe/tree/eltex-tv-fixes
> 
> It is far from proper adaptation, but at least all basic functionality (video playback, comments) is accessible via directional navigation.
> 
> In addition there is a special overlay that better highlights currently focused element.

I did not find any prebuild apk, so I try to build it on my own. With success.
Hope this was the right way to import the right branch? I choose the master:
![grafik](https://user-images.githubusercontent.com/12913436/72665648-e1c3a200-3a0a-11ea-8b8b-1ad98c61a5a2.png)
Then I build this app:
[app-debug.zip](https://github.com/TeamNewPipe/NewPipe/files/4080828/app-debug.zip)

Sadly there is nothing different to the current official stable release.
@Alexander-TX you mentioned that you add some dpad fixes. Can you tell me which git I can checkout to build this?

My problems with my nVidia ShieldTV (AndroidTV 9):
-no chance to control the video playback (pause, forward, backward, jump to a specific point)
-no chance to read comments
-difficult to play a clip with dpad on the remote/controller. Autoplay in landscape fullscrenn would be a good solution.

Try to setup an google-free androidtv system, newPipe feature is really necessary for this.
